### ISSUE TITLE
images on blog dont always appear

### ISSUE THREAD

**Issue by jakegalson** [devcows/hugo-universal-theme#93]
Often when I am on the blog page, the images next to the posts don't appear until I resize the page. Seems to me like this may be some kind of bug (at least in Safari).

**Comment by GeorgeWL** [comment]
Can you get a screen recorder of it?

**Comment by jakegalson** [comment]
I have put a test site here to see if the problem is replicated with you:
https://trucklab-testing.netlify.com

**Comment by GeorgeWL** [comment]
@jakegalson I don't have Safari unfortunately, can't see the same issue on other browsers.

**Comment by GeorgeWL** [comment]
Right, Just used https://browserling.com to test in on Safari, there doesn't seem to be an issue for me still.

Download a Screen Recorder app, open Safari Developer Console and the page, and then film what you get and upload it to something like Streamable.com  @jakegalson

**Comment by jakegalson** [comment]
It seems quite temperamental, and has worked the last few times I tried screen recording. Will see if I can capture anything, if not maybe the issue can be closed

**Comment by jakegalson** [comment]
Managed to get a screen recording:

https://streamable.com/fba4l

**Comment by GeorgeWL** [comment]
ooh okay. And annoyingly the console doesn't seem to give a reason either.

See my best guess is it's not playing nice with Bootstrap.js (which is the piece of JavaScript responsible for scaling the images) but I expected that to have something in the error log.

@adrianmo and @ryanfox1985 any ideas why this is happening for him?

**Comment by jakegalson** [comment]
It also happens on the iPad. But there you can't do the resize to get them to come up. Strange!

**Comment by GeorgeWL** [comment]
@jakegalson just as a test,  start one up using the default examplesite and see if it gets the same bug.

Then we can know whether it's the specific implementation or a bug across the whole theme.

**Comment by jakegalson** [comment]
It seems to be across the whole theme. I just tried it, and all of the images loaded apart from the last one (Creating a new theme). That one popped up when I resized the window.

**Comment by GeorgeWL** [comment]
One more test, to see if the specific Theme or if Hugo as a whole. 

Find another theme with a blog section, test that for us @jakegalson ?

Sorry to be asking you to do this, but I think you might be the only one we (the maintainers) know with a Mac atm.

Also btw, if you have the time, we could always do with an extra person to help with maintenance. Wanna join up?

**Comment by jakegalson** [comment]
I had a look at a couple of others, and there wasn't an issue. However, the problem seems oddly sporadic, so I will check back.

**Comment by jakegalson** [comment]
I just went on the site using Chrome in Windows - same problem.

**Comment by jakegalson** [comment]
I've created a new page (similar to the blog), and it has the same problem:
http://trucklab-testing.netlify.com/talk/

There is the same issue with the iPad. You either have to click on the page again to reload, or turn the iPad the other way to resize the page, at which point the images load.

**Comment by GeorgeWL** [comment]
@ringods you reckon can help here? I can't seem to replicate this.

**Comment by jakegalson** [comment]
This bug is still apparent, and I have seen it effect safari, chrome and edge so far.

**Comment by GeorgeWL** [comment]
so is it just ipad browser then? @jakegalson ?

**Comment by jakegalson** [comment]
its happened on and off on windows and mac

**Comment by jakegalson** [comment]
Did anyone manage to look into this any more?

**Comment by dudil** [comment]
Hi @jakegalson - I Got the root of this issue!
(Well not really an issue but still annoying)

It will happen when you are using netlify url without a custom domain:
Like in your example: http://trucklab-testing.netlify.com/talk/

But probably in your config.toml you are setting the baseURL to a different domain.

I had the same and chaining the baseURL into the netlify domain fixed it (until I will set it with my own custom domain)

So probably, in your case, setting in the config.toml:
baseURL=" http://trucklab-testing.netlify.com/"
Should fix this issue.

**Comment by jakegalson** [comment]
Hi @dudil 

Thanks for looking into it!

Unfortunately I don't think that is the problem though.

The site you are looking at is just one I was using for testing. The live version is here: https://trucklab.ch

In my config.toml, it is the following:
baseURL = "https://trucklab.ch/"

So they match up as far as I can tell

Jake

**Comment by dudil** [comment]
@jakegalson - No problem.
Just had the same issue and this solved it / also when I tried to publish on FB etc.

BTW - now browsing your site https://trucklab.ch/ everything is ok (I have Safari on Mac)
And the links you published earlier were of the netlify custom domain.

Are you sure this is not the issue you had?
Also - once you have the actual domain everything is taken from there...

Look at the attached screenshot - I was going through the netlify domain (https://trucklab.netlify.com/) you can see all resources are coming from your custom domain... (https://trucklab.ch)
![trucklab ch](https://user-images.githubusercontent.com/4785835/35406923-e66c9988-0212-11e8-8239-e7316150f8d2.png)

If you will put on your base URL another site you will see that the resources won't load (like no CSS and no images)

**Comment by mrowe** [comment]
I have been having this same problem. It is similarly infuriating intermittent and hard to debug.

I just tried two things, one of which seems to have helped (at least I have been unable to reproduce the problem since doing this):

 * change `baseurl` to be scheme relative (i.e. `//_hostname_/` without the `http`/`https`)
 * upgrade to bootstrap `3.3.7`

YMMV

**Comment by jakegalson** [comment]
@mrowe - I had a go at changing the baser to be relative. It hasn't fixed the issue completely, but it does seem to have made it better. I was already using bootstrap 3.3.7.

It seems once the image gets cached, it appears normally so it is only an issue the first time you visit the website. I get the feeling that something is pausing before the browser can load everything

**Comment by mostlyjason** [comment]
I looked at one of the missing images in inspector and saw that it was set to zero height. 

```
<div class="col-md-4">
    <div class="image" style="height: 0px;">                                    
        <a href="" target="_blank">
              <img src="image.gif" class="img-responsive" alt="">                    
         </a>
     </div>
</div>
```
In front.js there is a function called pictureZoom. If I run this function manually, the image suddenly appears and is set to a non-zero height. It appears either that this function isn't being called, or is being called before the image is loaded from the CDN. I added this to the top of my front.js file to call the function after page load. I can't reproduce the problem any more so I think it's solved?

```
$(window).on("load", function() {
  pictureZoom()
});
```

**Comment by mostlyjason** [comment]
FYI this looks similar to #142 which has a commit but I haven't tested that solution

**Comment by jakegalson** [comment]
Looks like this has fixed it! Thanks for looking into it.
